radix-engine/src/updates/cuttlefish.rs

```diff
@@ -194,6 +217,167 @@ fn generate_cuttlefish_transaction_validation_updates() -> StateUpdates {
     )
 }
 
+fn generate_cuttlefish_account_getters_extension_state_updates<S: SubstateDatabase + ?Sized>(
+    db: &S,
+) -> StateUpdates {
+    let reader = SystemDatabaseReader::new(db);
+    let node_id = ACCOUNT_PACKAGE.into_node_id();
+    let blueprint_version_key = BlueprintVersionKey {
+        blueprint: ACCOUNT_BLUEPRINT.to_string(),
+        version: Default::default(),
+    };
+
+    // Creating the original code substates for extension.
+    let (code_hash, (code_substate, vm_type_substate)) = {
+        let original_code = (NativeCodeId::AccountCode3 as u64).to_be_bytes().to_vec();
+
+        let code_hash = CodeHash::from_hash(hash(&original_code));
+        let code_substate = PackageCodeOriginalCodeV1 {
+            code: original_code,
+        }
+        .into_versioned()
+        .into_locked_substate();
+        let vm_type_substate = PackageCodeVmTypeV1 {
+            vm_type: VmType::Native,
+        }
+        .into_locked_substate();
+
+        (code_hash, (code_substate, vm_type_substate))
+    };
+
+    // Creating the new schema substate with the methods added by the extension
+    let (added_functions, schema) = AccountBlueprintCuttlefishExtension::added_functions_schema();
+    let (schema_hash, schema_substate) =
+        (schema.generate_schema_hash(), schema.into_locked_substate());
+
+    // Update the auth config of the account blueprint to have these added methods and have them be
+    // public.
+    let auth_config = {
+        let mut auth_config = AccountBlueprint::get_definition().auth_config;
+        let MethodAuthTemplate::StaticRoleDefinition(StaticRoleDefinition {
+            ref mut methods, ..
+        }) = auth_config.method_auth
+        else {
+            panic!("Account doesn't have a static role definition")
+        };
+        methods.extend(
+            added_functions
+                .keys()
+                .map(ToOwned::to_owned)
+                .map(|ident| MethodKey { ident })
+                .map(|key| (key, MethodAccessibility::Public)),
+        );
+        auth_config.into_locked_substate()
+    };
+
+    // Updating the blueprint definition of the existing blueprint with the added functions.
+    let blueprint_definition_substate = {
+        let mut blueprint_definition = reader
```

Comment:
I assume all the slight cost changes are because the blueprint definition is slightly bigger?

Reply:
Indeed. It's because there are more methods now added to the blueprint definition and also because the auth template is now slightly bigger with the three added methods. I remember mentioning this in Slack a while back and we agreed that this was indeed an issue but something we might want to change at a later point of time as it could only be changed by storing the blueprint definition in a different way.